### PR TITLE
Fix explore bug

### PR DESCRIPTION
**Pull request by chasulin**
Fixes: 
- The metric menu label wasn't updating when navigating to explore by clicking on a spark line
- We weren't properly resetting the explore default metric + default time range (and their respective menu labels) when navigating between location pages
- Share links are currently broken on prod ([example -- but beware it will freeze your tab](https://covidactnow.org/share/37933/?redirectTo=%2Fexplore%2F37933)) --- ([example of fixed share link from preview link](https://covid-projections-git-casulin-sparkline-expl-d59822-covidactnow.vercel.app/share/949/?redirectTo=%2Fexplore%2F949))

Also- consolidates `defaultMetric` and `currentMetric` into `currentMetric` (and moves `currentMetric` to the top level)

### CODE REVIEW

File: src/components/Explore/Explore.tsx [hunk 1]
```diff
@@ -160,12 +160,30 @@ const ExploreCopy: React.FunctionComponent<{
...
 
+    const [period, setPeriod] = useState<Period>(Period.ALL);
+    const allPeriodLabels = getAllPeriodLabels();
+    const dateRange = getDateRange(period);
+
+    const onSelectPeriod = (newPeriod: Period) => {
+      const newPeriodLabel = periodMap[newPeriod].label;
+      setPeriod(newPeriod);
+      trackExploreEvent(EventAction.SELECT, `Period: ${newPeriodLabel}`);
+    };
```
Comment (chasulin) [inline src/components/Explore/Explore.tsx:171]: nothing above this point has changed, just moved it all up on the page

File: src/components/Explore/Explore.tsx [hunk 2]
```diff
@@ -227,15 +237,20 @@ const ExploreCopy: React.FunctionComponent<{
...
+    // default metric setting
+    // (We also reset the time range to 'all time')
+    useEffect(() => {
+      if (pathname.includes('/explore')) {
+        const timeoutId = scrollToExplore();
+        return () => clearTimeout(timeoutId);
+      }
+      setCurrentMetric(ExploreMetric.CASES);
+      setPeriod(Period.ALL);
+    }, [pathname, scrollToExplore]);
```
Comment (chasulin) [inline src/components/Explore/Explore.tsx:253]: just an edit to an existing useEffect (prev on line 200 in the orig part of the diff) -- adds:
```
setCurrentMetric(ExploreMetric.CASES);
setPeriod(Period.ALL);
```

(explanation in the added comment^^)

File: src/components/Explore/Explore.tsx [hunk 3]
```diff
@@ -160,12 +160,30 @@ const ExploreCopy: React.FunctionComponent<{
...
-    }, [currentMetric, selectedLocations]);
+      setMetricMenuLabel(metricLabels[currentMetric]);
+      setTimeRangeMenuLabel(allPeriodLabels[period]);
+    }, [
+      currentMetric,
+      selectedLocations,
+      metricLabels,
+      allPeriodLabels,
+      period,
+    ]);
```
Comment (pnavarrc) [inline src/components/Explore/Explore.tsx:186]: How about defining variables and setting them as dependencies instead?

```tsx
const timeRangeLabel = allPeriodLabels[period];
const currentMenuLabel = metricLabels[currentMetric];
const isNormalizedData = selectedLocations.length > 1 && /* ... */;

useEffect(() => {
  /* update the state using the values */
}, [timeRangeLabel, isNormalized, currentMenuLabel]);
```

I'm not sure if this is actually better, but I feel that having fewer dependencies might help to update the state only when needed

**Comment by chasulin** [comment]
@pnavarrc we previously had `defaultMetric`/`setDefaultMetric` (used to set tab when clicking spark line) **and** `currentMetric`/`setCurrentMetric` (used for changing metric with the dropdown menu)-- i got rid of `defaultMetric`/`setDefaultMetric` and consolidated into `currentMetric`/`setCurrentMetric`. i think this solves the weird bugs like the one you sent on slack

File: src/components/Explore/Explore.tsx [hunk 4]
```diff
@@ -267,9 +291,13 @@ const ExploreCopy: React.FunctionComponent<{
...
         );
         setSelectedLocations(locations);
-        setMetricMenuLabel(metricLabels[sharedParams.currentMetric]);
+        setRegionNamesMenuLabel(locations.map(getLocationLabel).join('; '));
       }
-    }, [sharedParams, metricLabels]);
+    }, [setCurrentMetric, sharedParams]);
+
+    useEffect(() => {
+      setMetricMenuLabel(metricLabels[currentMetric]);
```
Comment (chasulin) [inline src/components/Explore/Explore.tsx:299]: moving this setState into its own effect (and out of the one above) seems to have fixed the broken share links

File: src/components/Explore/Explore.tsx [hunk 5]
```diff
@@ -160,12 +163,34 @@ const ExploreCopy: React.FunctionComponent<{
...
 
+    const [period, setPeriod] = useState<Period>(Period.ALL);
+    const allPeriodLabels = getAllPeriodLabels();
+    const dateRange = getDateRange(period);
+
+    const onSelectPeriod = (newPeriod: Period) => {
+      const newPeriodLabel = periodMap[newPeriod].label;
+      setPeriod(newPeriod);
+      trackExploreEvent(EventAction.SELECT, `Period: ${newPeriodLabel}`);
+    };
```
Comment (chasulin) [inline src/components/Explore/Explore.tsx:174]: just moved these two blocks up on the page^

File: src/components/Explore/Explore.tsx [hunk 6]
```diff
@@ -279,6 +291,11 @@ const ExploreCopy: React.FunctionComponent<{
     const showLegend =
       ORIGINAL_EXPLORE_METRICS.includes(currentMetric) && numLocations === 1;
 
+    // menu labels for metric, time period, and selected locations:
+    const metricMenuLabel = metricLabels[currentMetric];
+    const regionsMenuLabel = selectedLocations.map(getLocationLabel).join('; ');
+    const periodMenuLabel = allPeriodLabels[period];
```
Comment (chasulin) [inline src/components/Explore/Explore.tsx:297]: these 3 were all previously state variables ^

**Review by pnavarrc — APPROVED** [review]
Looks good! We should move the remaining state to the location page next week

**Pull request merged**